### PR TITLE
Add new datalayer value to track blocks

### PR DESCRIPTION
**Pull request by GP-Dan-Tovbein**
Ref: https://jira.greenpeace.org/browse/PLANET-7075

## Description
Create a new datalayer value to track Planet4 blocks and Gravity forms

## Testing
Create a new page with p4 or gravity form blocks and preview it.
Then, open de browser console and type `datalayer` through the terminal. Expand the `zero` index and check the value of `p4_blocks`. All the blocks should be separated by commas.

Screenshot 2023-03-21 at 16 09 38

### CODE REVIEW

File: src/Context.php
Comment (GP-Dan-Tovbein) [inline src/Context.php:118]: Example of string `columns,gravityforms/form`

File: src/Context.php
Comment (GP-Dan-Tovbein) [inline src/Context.php:113]: Take only `gravityforms` from `wp:gravityforms/form`

**Comment by mleray** [comment]
Don't we need UAT for this ticket? 🤔 Maybe Molly or Julia should check it out

**Comment by GP-Dan-Tovbein** [comment]
> Don't we need UAT for this ticket? 🤔 Maybe Molly or Julia should check it out... Either way, could you please set up a test instance for this? On local it doesn't work for me 😕

Probably because you haven't set the `Google Tag Manager Container` through `Planet 4 > Analytics`. By the way, the umbriel instance is ready :)

File: src/Context.php
Comment (GP-Dan-Tovbein) [inline src/Context.php:106]: This regex matches all the occurrences of `wp:planet4-blocks/` or `wp:gravityforms/` until the next empty space. 

For example, 
`wp:planet4-blocks/columns`, `wp:planet4-blocks/cover`, `wp:gravityforms/form`

**Comment by mleray** [comment]
On this [post](https://www-dev.greenpeace.org/test-umbriel/press/1111/duis-posuere-5/) there is the `Related Articles` block, but it doesn't show in the `p4_blocks`, not sure that's expected? I know it's a bit different since it's not technically part of the post content 🤔

File: src/Context.php
Comment (mleray) [inline src/Context.php:104]: Why is there a `&` in front of the `$context` param?

**Review by mleray — COMMENTED** [review]
Code looks good to me 👏 I only have one comment: on posts such as [this one](https://www-dev.greenpeace.org/test-umbriel/press/1111/duis-posuere-5/) there is the `Related Articles` block, but it doesn't show in the `p4_blocks`, not sure that's expected? I know it's a bit different since it's not technically part of the post content 🤔

File: single.php
Comment (GP-Dan-Tovbein) [inline single.php:128]: Added before the render to make sure that the `$post` has included the right content.

**Comment by GP-Dan-Tovbein** [comment]
@mleray FYI https://jira.greenpeace.org/browse/PLANET-7075?focusedId=247673&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-247673

**Review by mleray — APPROVED** [review]
Code looks good to me 👏 and it seems to work as expected, but let's wait for Molly's approval before merging 🙂

**Pull request merged**
